stream-chat-android-client/src/main/java/io/getstream/chat/android/client/socket/SocketFactory.kt

```diff
@@ -114,5 +114,11 @@ internal class SocketFactory(
         ) : ConnectionConf()
 
         internal fun asReconnectionConf(): ConnectionConf = this.also { isReconnection = true }
+
+        internal val id: String
+            get() = when (this) {
+                is AnonymousConnectionConf -> user.id.replace("!", "")
```

Comment:
It is true that we know that the `!` will be always the first position, but removing the first character could be more confusing than replacing the illegal character. The anonymous `id` is only 5 characters, I don't think it will affect the performance at all